format/FlightSql.proto

```diff
@@ -867,6 +867,69 @@ enum SqlSupportsConvert {
+ *                     For string data, this is the length in characters.
+ *                     For datetime data types, this is the length in characters of the string representation.
+ *                     NULL is returned for data types where column size is not applicable.),
+ *   literal_prefix: utf8 (Character or characters used to prefix a literal, NULL is returned for
+ *                         data types where a literal prefix is not applicable.),
+ *   literal_suffix: utf8 (Character or characters used to terminate a literal,
+ *                         NULL is returned for data types where a literal suffix is not applicable.),
+ *   create_params: utf8 (A list of keywords, separated by commas, corresponding
+ *                        to each parameter that the application may specify in parentheses
+ *                        when using the name that is returned in the TYPE_NAME field.
+ *                        NULL is returned if there are no parameters for the data type definition.),
+ *   nullable: int not null (Shows if the data type accepts a NULL value),
+ *   case_sensitive: bool not null (Shows if a character data type is case-sensitive in collations and comparisons),
+ *   searchable: int not null (Shows how the data type is used in a WHERE clause),
+ *   unsigned_attribute: bool (Shows if the data type is unsigned. NULL is returned if the attribute is
+ *                             not applicable to the data type or the data type is not numeric.),
+ *   fixed_prec_scale: bool not null (Shows if the data type has predefined fixed precision and scale.),
+ *   auto_increment: bool (Shows if the data type is auto incremental. NULL is returned if the attribute
+ *                         is not applicable to the data type or the data type is not numeric.),
+ *   local_type_name: utf8 (Localized version of the data source-dependent name of the data type. NULL
+ *                          is returned if a localized name is not supported by the data source),
+ *   minimum_scale: int (The minimum scale of the data type on the data source.
+ *                       If a data type has a fixed scale, the MINIMUM_SCALE and MAXIMUM_SCALE
+ *                       columns both contain this value. NULL is returned if scale is not applicable.),
+ *   maximum_scale: int (The maximum scale of the data type on the data source.
+ *                       NULL is returned if scale is not applicable.),
+ *   sql_data_type: int not null (The value of the SQL data type as it appears in the
+ *                                SQL_DESC_TYPE field of the descriptor. This column is
+ *                                the same as the DATA_TYPE column, except for interval and
+ *                                datetime data types.),
+ *   sql_datetime_sub: int (When the value of SQL_DATA_TYPE is SQL_DATETIME or SQL_INTERVAL,
+ *                          this column contains the datetime/interval subcode. For data types
```

Comment:
What is the subcode?

Reply:
Ah, it appears to pair with sql_data_type above.

Reply:
Could it be written out as `datetime_subcode` or something? (The `sql_` prefix on these field names seems generally redundant.)

Reply:
It seems ODBC has defined values for this, can we have an enum for them?

Reply:
Fixed